### PR TITLE
(PUP-4890) Add code_id to the catalog

### PR DESCRIPTION
**Pull request by joshcooper**
Adds a code_id accessor to the catalog, updates the api docs, schema, and tests.

Updates the compiler terminus' find method to add the code_id from the request to the catalog it returns.

### CODE REVIEW

**Comment by joshcooper** [comment]
@kylog @cprice404 @lutter @senior I hear you want a code_id?

**Comment by kylog** [comment]
:+1:

File: lib/puppet/indirector/catalog/compiler.rb [hunk 1]
```diff
@@ -48,6 +48,9 @@ def find(request)
     node.trusted_data = Puppet.lookup(:trusted_information) { Puppet::Context::TrustedInformation.local(node) }.to_h
 
     if catalog = compile(node)
+      if code_id = request.options[:code_id]
```
Comment (lutter) [inline lib/puppet/indirector/catalog/compiler.rb]: Sorry for not seeing this earlier: it looks like with this setup, `code_id` is not available during compilation, though we need that so we can query PDB with it when we look up consumed resources.

Comment (joshcooper) [inline lib/puppet/indirector/catalog/compiler.rb]: That's a bummer. @lutter can you point me to code where you need access to it?

Comment (lutter) [inline lib/puppet/indirector/catalog/compiler.rb]: Left a comment on PUP-4890 explaining that

**Pull request merged**